lib/common/http2client.c

```diff
@@ -1111,6 +1115,10 @@ static void on_write_complete(h2o_socket_t *sock, const char *err)
     /* reset the other buffer */
     h2o_buffer_dispose(&conn->output.buf_in_flight);
 
+    /* as request write, unlink the deferred timeout that might have been set by `proceed_req` called above */
```

Comment:
That's a good point.

I think what we have now is fine, because `do_emit_writereq` (invoked from `on_notify_write`) would be a no-op when `conn->output.buf->size != 0 || !h2o_linklist_is_empty(&conn->output.sending_streams)`. Therefore, we have no reason to call `h2o_socket_notify_write`.

That said, the code is confusing because we call `do_emit_writereq` in case of libuv, without checking the same condition.

I've tightened the code in 189ac93, PTAL.

Reply:
The code looks simpler and easier to understand now 👍

Reply:
Thank you for checking!